compose-builder/tui-generator.sh

```diff
@@ -0,0 +1,299 @@
+function displayNoSecty() {
+    message="Would you like to generate a non-secure configuration of the Compose file?"
+    $WHIPTAIL --title "Non-Secure Configuration" --yesno "$message" $LINES $COLUMNS
+}
+
+
+####################################################################
+#    Device Services Options Display Function                      #
+####################################################################
+function devServiceOption() {
+    message="Available Device Services to include in your compose file.\n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${deviceServices[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${deviceServiceDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_DEVSERVICES=$($WHIPTAIL --title "Device Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --checklist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+####################################################################
+#    Message Bus Options Display Function                          #
+####################################################################
+function msgBusOption() {
+    message="Available Message Buses to replace the default one.\n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${messageBus[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${msgBusDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_BUS=$($WHIPTAIL --title "App Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --radiolist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+####################################################################
+#    Generate / Generate & Run Option Display Function             #
+####################################################################
+function finalStep() {
+    message="What would you like to do?"
+    $WHIPTAIL --title "Non-Secure Configuration" \
+        --yes-button "Generate File" \
+        --no-button "Generate File and Run" \
+         --yesno "$message" $LINES $COLUMNS
+}
+
+## Step - 1: Start by Asking about whether Images should be pulled for ARM64?
+displayArm64
+if [[ $? == 0 ]]; then
+    SELECTED_OTHERS+=("arm64 ")
```

Comment:
Instead of lumping these in with the others, I would probably just have a specific one:
```shell
ARM64_OPT=""
if displayArm64; then
    ARM64_OPT="arm64"
fi
```

then, in your make gen, just reference it with `"${ARM64_OPT}"`. If it is blank, nothing with be added to the command.

Reply:
I tried your suggestion however, one problem that occurs when using `"${SECURITY_OPT}"` with the following:

    make gen "${ARM_64_OPT}" "${SECURITY_OPT}"

is the fact that when both are left blank (not-selected) then `make` throws the following error:

    make: *** empty string invalid as file name.  Stop.

I suspect that two empty spaces leads before this leads to such a problem. A potential solution is just to resolve the variables without the double quotes `"`

    make gen ${ARM64_OPT} ${SECURITY_OPT}